pkg/grafana/datasource_test.go

```diff
@@ -2,39 +2,46 @@ package grafana
 
 import (
 	"encoding/json"
+	gapi "github.com/grafana/grafana-api-golang-client"
 	"os"
 	"testing"
 
 	"github.com/grafana/grizzly/pkg/grizzly"
 	"github.com/stretchr/testify/require"
 )
 
-func TestDatasources(t *testing.T) {
+func TestDataSources(t *testing.T) {
 	os.Setenv("GRAFANA_URL", getUrl())
 
+	client, err := gapi.New(getUrl(), gapi.Config{})
+	require.NoError(t, err)
+
 	grizzly.ConfigureProviderRegistry(
 		[]grizzly.Provider{
-			&Provider{},
+			&Provider{client: client},
 		})
 
 	ticker := pingService(getUrl())
 	defer ticker.Stop()
 
+	handler, err := grizzly.Registry.GetHandler((&DatasourceHandler{}).Kind())
+	require.NoError(t, err)
+
 	t.Run("get remote datasource - success", func(t *testing.T) {
-		resource, err := getRemoteDatasource("AppDynamics")
+		resource, err := handler.GetByUID("AppDynamics")
 		require.NoError(t, err)
 
 		require.Equal(t, resource.APIVersion(), "grizzly.grafana.com/v1alpha1")
 		require.Equal(t, resource.Name(), "AppDynamics")
-		require.Len(t, resource.Spec(), 18)
+		require.Len(t, resource.Spec(), 13)
```

Comment:
Zero values are now ignored (`database`, `jsonData`, `secureJsonFields`, etc), but the behavior should remain the same, as these are non-required by the HTTP API (so once `apply`-ed back, should be equivalent).